core/src/validator.rs

```diff
@@ -119,6 +119,51 @@ use {
 const MAX_COMPLETED_DATA_SETS_IN_CHANNEL: usize = 100_000;
 const WAIT_FOR_SUPERMAJORITY_THRESHOLD_PERCENT: u64 = 80;
 
+pub const DEFAULT_REPLAYING_BACKEND: &str = "blockstore-processor";
+pub const DEFAULT_BANKING_BACKEND: &str = "multi-iterator";
+
+#[derive(Clone, Debug)]
+pub enum ReplayingBackend {
+    BlockstoreProcessor,
+    UnifiedScheduler,
+}
+
+impl Default for ReplayingBackend {
+    fn default() -> Self {
+        Self::from(DEFAULT_REPLAYING_BACKEND)
+    }
+}
+
+impl From<&str> for ReplayingBackend {
+    fn from(string: &str) -> Self {
+        match string {
+            "blockstore-processor" => Self::BlockstoreProcessor,
+            "unified-scheduler" => Self::UnifiedScheduler,
+            bad_backend => panic!("Invalid replaying backend: {bad_backend}"),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum BankingBackend {
```

Comment:
... also i avoided the use of work `stage`. it's mismatch in the context of `solana-ledger-tool`.